### ISSUE TITLE
Parallel gateway and missing task data

### ISSUE THREAD

**Issue by sashayar13** [sartography/spiff-arena#1502]
There was an issue in prod where task data didn’t go through a merging parallel gateway.

The currency_enum data was produced inside a call activity Get currency list but was not available after the merging parallel gateway. You can see more details in the attached screenshots (https://www.notion.so/Parallel-gateway-and-missing-task-data-b55d50b7d29c422f95de834da9726052?pvs=4)

**Comment by madhurrya** [comment]
I have not noticed it in any environment other than in dev.mod.
Do you think the fix you did for the dev.mod issue will fix this as well?
It's the same issue but in dev.mod it was consistently happening, but here it is Random and we never noticed it during our testing.

**Comment by jasquat** [comment]
Yeah that is odd. I do think the same fix would fix it. I'm not sure how this could happen without celery if it doesn't happen in non-prod. How often has this happened?

**Comment by madhurrya** [comment]
I think just once so far. But it has been just one day since we deployed the latest to Prod.

**Comment by calexh-sar** [comment]
On hold pending additional occurrences and/or being able to reproduce.

**Comment by sashayar13** [comment]
just reproduced this in prod again (and didn't notice this behaviour in test.app or dev.app)
Task data from the parallel paths doesn't come through to the Compare task

![Image](https://github.com/sartography/spiff-arena/assets/104367255/584f391a-0073-48dc-88f0-f03d3ef3a100)

Got the error (also claiming that the task data didn't come through, however it should) in dev.app after the process modification to:
https://dev.app.spiff.status.im/i/6889

![Image](https://github.com/sartography/spiff-arena/assets/104367255/168eb9b4-54a8-4828-8d6a-e147f758c1ee)

process model is https://dev.app.spiff.status.im/process-models/misc:data-sync:bbhr-data-sync:orgstructure-bbhr-iplicit

**Comment by sashayar13** [comment]
the process without the parallel gateway worked just fine in prod - no errors

**Comment by madhurrya** [comment]
@sashayar13 I was trying to recreate this issue using a simple model and when creating it I have mistakenly added the closing gateway as a exclusive instead of parallel and I got the same error as you did. And later realized my mistake and when I checked your diagram above noticed that you have also done the same :-) So it must be the reason why you got it in dev.app
![image](https://github.com/sartography/spiff-arena/assets/110527567/8b7e1d6a-9e9d-4e93-a2e7-76b144f8a463)

When I changed my model correctly to parallel gateway, it works fine. https://dev.app.spiff.status.im/process-models/misc:qa:bpmn-model-testing:unit-test-parallel-gateway-2

What is the process that you get this error in Prod? Will try to create a similar simpler one to recreate it.

**Comment by jasquat** [comment]
Oh, I wonder if it's related to the manual tasks after the call activities / subprocesses. Was that first screenshot taken from prod or dev.app?

**Comment by sashayar13** [comment]
@jasquat another occurance with the currency_enum list occured in prod https://prod.mod.spiff.status.im/i/6487
Upload expense evidence - travel process

**Comment by sashayar13** [comment]
the process in dev - https://dev.app.spiff.status.im/process-models/manage-finance:accounts-payable:process-expense-reports:upload-expense-evidence-travel
The process is started by a Message event

Something I noticed is that in both cases, in prod.mod - The issue appeared when I had to manually restart the Get Supplier details call activity because of the Iplicit issues. https://dev.app.spiff.status.im/editor/process-models/manage-finance:accounts-payable:process-expense-reports:capture-expense-data/files/capture-expense-details.bpmn
The steps I follow to debug Iplicit related issues:
- Rewind the process to the start event of the call activity
- Resume the process

![Image](https://github.com/sartography/spiff-arena/assets/104367255/fa50dfb4-b467-4fe9-b930-ae274775bb6f)

**Comment by burnettk** [comment]
we created a process model with two call activities run in parallel, like the last screenshot with Get Xe Currencies Enum List and Get Supplier details. We put a service task inside the bottom one that was designed to fail. After it failed, we rewinded to the beginning of the bottom called activity and ran it again (making the service task pass this time). the task data generated from the top and bottom call activities was present at the parallel gateway and in subsequent tasks. so we failed to repro, alas.
